src/main/java/dev/foxikle/customnpcs/internal/listeners/Listeners.java

```diff
@@ -12,35 +13,160 @@
 import net.minecraft.world.phys.Vec3;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
+import org.bukkit.World;
 import org.bukkit.craftbukkit.v1_20_R2.entity.CraftPlayer;
 import org.bukkit.entity.Entity;
 import org.bukkit.entity.EntityType;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
+import org.bukkit.scheduler.BukkitRunnable;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.bukkit.event.player.*;
+import org.bukkit.command.ConsoleCommandSender;
 import org.bukkit.inventory.EquipmentSlot;
 
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.UUID;
 
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import java.util.regex.Pattern;
 /**
  * The class that deals with misc listeners
  */
 public class Listeners implements Listener {
+	/**
+	 * Player Movement Data that keeps track of old movements to replace PlayerMoveEvent
+	 * @since *Insert_Version*
+	 */
+	private static final ConcurrentMap<UUID, MovementData> playerMovementData = new ConcurrentHashMap<>();
+	
+	// Helper Constants
+	// since *Insert_Version*
+	private static final int FIVE_BLOCKS = 25;
+	private static final int FIFTY_BLOCKS = 2500; // 50 * 50
+	private static final int FOURTY_BLOCKS = 2304; // 48 * 48
+	private static final double HALF_BLOCK = 0.25;
+	
+	// Writing Constants
+	// since *Insert_Version*
```

Comment:
Correct me if I'm wrong, but I don't think a since is really supposed to be here